node_client/node/node/node.py

```diff
@@ -78,3 +74,47 @@ def submit_comment():
 @app.route('/pin/manuscript', methods=["GET"])
 def pin_manuscript():
     pass
+
+@app.route('/upload/file', methods=['GET','POST'])
+def upload_file():
+    if request.method == 'POST':
+        print(request.files)
+        # check if the post request has the file part
+        if 'paper' not in request.files:
+            flash('No file part')
+            file_present = False
+        file = request.files['paper']
+        # if user does not select file, browser also
+        # submit a empty part without filename
+        if file.filename == '':
+            flash('No selected file')
+            file_present = False
+        if file and allowed_file(file.filename):
+            #make an upload dir if it doesn't exist
+            folder = pyx.arxain_path+'/upload'
+            if not os.path.exists(folder):
+                    os.makedirs(folder)
+            #delete files already in upload folder
+            for the_file in os.listdir(folder):
+                file_path = os.path.join(folder, the_file)
+                try:
+                    if os.path.isfile(file_path):
+                        os.unlink(file_path)
+                except Exception as e:
```

Comment:
Can we find a specific `Exception` for this (probably an `IOError`, but not sure)? It will help with future debugging and testing.

Reply:
This was code written by @bhaney that I moved from pyXain to node, and am a little un-sure about what could go wrong here so hopefully he can help out on this. This is also stuff that will only really be necessary for when we have people use our own server as this is a post-method handling the uploading of a pdf which we don't need if the user is running a local node.

Reply:
so right now the contract
>an author cannot revise their own paper
An author can revise their paper (right?, we want to be able to submit new versions)
>a non-author cannot revise someone else's paper
So right now the contract allows the submitter to add other authors. They can revise the paper too. But these are the only people
>revisions had to actually be revisions
Yea, this is also in the contract. It has to be a different paper IPFS hash